pkgs/tools/misc/trackma/default.nix

```diff
@@ -19,33 +33,49 @@ python3.pkgs.buildPythonApplication rec {
     repo = "trackma";
     rev = "v${version}";
     sha256 = "sha256-drc39ID4WYBQ/L2py57CB5OkQNfRKNigPQW0Lp8GIMc=";
+    fetchSubmodules = true; # for anime-relations submodule
   };
 
-  nativeBuildInputs = lib.optionals withGtk [ wrapGAppsHook ];
+  nativeBuildInputs = lib.optionals withGtk [ wrapGAppsHook ]
+    ++ lib.optionals withQt [ qt5.wrapQtAppsHook ];
 
   buildInputs = lib.optionals withGtk [ glib gobject-introspection gtk3 ];
 
-  propagatedBuildInputs = [ python3.pkgs.urllib3  python3.pkgs.dbus-python ]
-    ++ lib.optionals withGtk [ python3.pkgs.pillow python3.pkgs.pygobject3 python3.pkgs.pycairo ]
+  propagatedBuildInputs = [ python3.pkgs.urllib3 ]
+    ++ lib.optionals withQt [ python3.pkgs.pyqt5 ]
+    ++ lib.optionals withGtk [ python3.pkgs.pycairo ]
     ++ lib.optionals withCurses [ python3.pkgs.urwid ]
-    ++ lib.optionals stdenv.isLinux [ python3.pkgs.pyinotify ];
+    ++ lib.optionals stdenv.isLinux [ python3.pkgs.dbus-python python3.pkgs.pygobject3 python3.pkgs.pyinotify ]
+    ++ lib.optionals (withGtk || withQt) [ python3.pkgs.pillow ];
 
   # broken with gobject-introspection setup hook, see https://github.com/NixOS/nixpkgs/issues/56943
   strictDeps = false;
 
-  dontWrapGApps = true; # prevent double wrapping
+  dontWrapQtApps = true;
+  dontWrapGApps = true;
 
   preFixup = ''
-    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+    ${lib.optionalString withQt "wrapQtApp $out/bin/trackma-qt"}
+    ${lib.optionalString withGtk "wrapGApp $out/bin/trackma-gtk"}
+  '';
+
+  desktopItemQt = mkDesktopItem "trackma-qt" "Trackma (Qt)" "Trackma Updater (Qt-frontend)" false;
+  desktopItemGTK = mkDesktopItem "trackma-gtk" "Trackma (GTK)" "Trackma Updater (Gtk-frontend)" false;
+  desktopItemCurses = mkDesktopItem "trackma-curses" "Trackma (ncurses)" "Trackma Updater (ncurses frontend)" true;
+
+  postInstall = ''
+    install -Dvm444 $src/$pname/data/icon.png $out/share/pixmaps/$pname.png
+    ${lib.optionalString withQt "install -Dvm444 $desktopItemQt/share/applications/* -t $out/share/applications/"}
+    ${lib.optionalString withGtk "install -Dvm444 $desktopItemGTK/share/applications/* -t $out/share/applications/"}
+    ${lib.optionalString withCurses "install -Dvm444 $desktopItemCurses/share/applications/* -t $out/share/applications/"}
   '';
 
   doCheck = false;
 
   pythonImportsCheck = [ "trackma" ];
 
-  # FIXME(trackma-qt): https://github.com/NixOS/nixpkgs/pull/179715#issuecomment-1171371059
   postDist = ''
-    rm $out/bin/trackma-qt
+    ${lib.optionalString (!withQt) "rm $out/bin/trackma-qt"}
```

Comment:
The same advice about string interpolation.

Reply:
I think I broke my record of mistakes in a single PR. :face_in_clouds:

Reply:
Perfectly normal for beginners. After some months and years practicing it will become natural.